clcache.py

```diff
@@ -142,7 +218,13 @@ def __init__(self):
             os.makedirs(self.objectsDir)
         lockName = self.cacheDirectory().replace(':', '-').replace('\\', '-')
         timeout_ms = int(os.environ.get('CLCACHE_OBJECT_CACHE_TIMEOUT_MS', 10 * 1000))
-        self.lock = ObjectCacheLock(lockName, timeout_ms)
+
+        cfg = Configuration(self)
```

Comment:
This one is tricky ;) We need to hold a lock before we're allowed to access the configuration. At least this is what the current code suggests: https://github.com/frerich/clcache/blob/dc63c22f2d36fc624ea83f0187ef045e09c2c0e4/clcache.py#L1104

So this might be the point where moving to environment variables makes life easier.